### PR TITLE
Tweak error codes according to the conformance suite

### PR DESCRIPTION
**Pull request by srikrsna-buf**
Tweak error codes according to the conformance suite. Users should not notice any difference as most of the tweaks are related to errors caused due to mis-implementation of the protocol.

Also update conformance runner to `v1.0.2`. This lets us shed `node-forge` and `asn1js` which we relied to generate a TLS cert.

### CODE REVIEW

File: packages/connect-web-test/src/conformance/unimplemented_server_streaming_method.spec.ts [hunk 1]
```diff
@@ -22,7 +22,7 @@ import { TestService } from "../gen/connectrpc/conformance/v1/test_connect.js";
 import { describeTransports } from "../helpers/conformanceserver.js";
 import { Empty } from "@bufbuild/protobuf";
 
-describe("unimplemented_server_streaming_method", function () {
+xdescribe("unimplemented_server_streaming_method", function () {
```
Comment (smaye81) [inline packages/connect-web-test/src/conformance/unimplemented_server_streaming_method.spec.ts:25]: Is this meant to be skipped (along with the callback client one above?)

File: packages/connect-web/src/connect-transport.ts [hunk 2]
```diff
@@ -268,6 +275,12 @@ export function createConnectTransport(
             break;
           }
           const { flags, data } = result.value;
+          if ((flags & compressedFlag) === compressedFlag) {
+            throw new ConnectError(
+              `protocol error: recevied unsupported compressed output`,
```
Comment (smaye81) [inline packages/connect-web/src/connect-transport.ts:280]: nit: `received...`. (here and elsewhere in a few places)

File: packages/connect/src/promise-client.ts [hunk 3]
```diff
@@ -163,13 +163,21 @@ export function createClientStreamingFn<
...
     if (!singleMessage) {
       throw new ConnectError(
         "protocol error: missing response message",
-        Code.Internal,
+        Code.Unimplemented,
+      );
+    }
+    if (count > 1) {
+      throw new ConnectError(
+        "protocol error: received extra messages for client steaming method",
```
Comment (smaye81) [inline packages/connect/src/promise-client.ts:179]: nit: 'streaming'

**Review by smaye81 — COMMENTED** [review]
Just a thought, but is it worth splitting up the actual changes to the Connect library into their own PR(s) just to able to reason about the fixes a bit better?

**Comment by srikrsna-buf** [comment]
> Just a thought, but is it worth splitting up the actual changes to the Connect library into their own PR(s) just to able to reason about the fixes a bit better?

Then we'd end up not having tests for that, or we can update the conformance version first by adding a bunch of know failing, and then add the fixes removing the know failing cases.

It didn't seem worth it because the update is just a version bump and deleting some files, and all the fixes are status code related.

File: packages/connect-web-test/src/conformance/unimplemented_server_streaming_method.spec.ts (hunk 1 above)
Comment (srikrsna-buf) [reply packages/connect-web-test/src/conformance/unimplemented_server_streaming_method.spec.ts:25]: Yes, this fails for both

**Comment by smaye81** [comment]
Seems fine to me, but let's wait to see what @timostamm thinks.

**Review by smaye81 — APPROVED** [review]

File: packages/connect-conformance/package.json [hunk 4]
```diff
@@ -23,8 +27,6 @@
     "@bufbuild/protobuf": "^1.7.2",
     "@connectrpc/connect": "1.4.0",
     "@connectrpc/connect-node": "1.4.0",
-    "node-forge": "^1.3.1",
-    "asn1js": "^3.0.5",
```
Comment (timostamm) [inline packages/connect-conformance/package.json:27]: 🎉

**Review by timostamm — COMMENTED** [review]
If we cut a release, this PR shows up in the changelog, and the title should tell users what we are changing.

The PR title doesn't give any indication that we're changing error codes, and it's not clear what exactly changes from looking at the diff.

We need better communication. For reference, see https://github.com/connectrpc/connect-go/pull/706.

I suggest to change the PR title to be about the change to the code mappings, and only mention the conformance update in the description. You can link to the connectrpc.com PR, same as the connect-go change did. If there are other changes beyond the status code changes in this PR, please list them as well.

**Review by timostamm — APPROVED** [review]

**Pull request merged**